feddlib/core/FE/FiniteElement.hpp

```diff
@@ -107,6 +107,28 @@ class FiniteElement {
 
 	bool isMarkedEdge(){return markedEdge_;};
 
+    // Computed in Domain_def.hpp -> Save in each (Sub) Finite Element the outward normal if the Finite Element is laying on the outer boundary
+    void setSurfaceNormal(vec_dbl_Type  computedNormal){ this->surfaceNormal =  computedNormal; };
+
+    // Return the vector saved for the surface Normal of the edge/ face
+    vec_dbl_Type getSurfaceNormal()
+        { if (this->surfaceNormal.size() == 0){ TEUCHOS_TEST_FOR_EXCEPTION(true,std::runtime_error,"SurfaceNormal is not initialized")};
+        return this->surfaceNormal;};
+
+    // Getter and setter functions for a surface element
+    void setNeumannBC_element(bool flag){ this->neumannBC_element = flag ; };   
+    bool getNeumannBC_element(){ return this->neumannBC_element; };   
+    void setElement_Scaling(double elscaling){ this->elScaling = elscaling ; };   
+    double getElement_Scaling(){ return this->elScaling ; };   
+    void setQuadratureWeightsRef(vec_dbl_Type weights){this->neumannBC_QuadratureWeightsRef = weights;};
+    void setQuadraturePointsPhy(vec2D_dbl_Type qpoints){this->neumannBC__QuadraturePointsPhy = qpoints; };
+    vec_dbl_Type getQuadratureWeightsRef()
+        { if (this->neumannBC_QuadratureWeightsRef.size() == 0){ TEUCHOS_TEST_FOR_EXCEPTION(true,std::runtime_error,"neumannBC_QuadratureWeightsRef is not initialized")};
+        return this->neumannBC_QuadratureWeightsRef;};
+    vec2D_dbl_Type getQuadraturePointsPhy()
+        { if (this->neumannBC__QuadraturePointsPhy.size() == 0){ TEUCHOS_TEST_FOR_EXCEPTION(true,std::runtime_error,"neumannBC__QuadraturePointsPhy is not initialized")};
```

Comment:
Is this supposed to be Phi?

Reply:
I will rename it to neumannBCQuadraturePointsGlobal - these are the quadrature points defined on the global coordinate system - They will be mapped back to the reference line/ surface in AssembleFESpecific Classes -> why ? because how do we now which boundary surface will be mapped onto which Reference edge/ face therefore it is easier to just compute them in global coordinate system